FirebaseStorage/Sources/StorageReference.swift

```diff
@@ -705,4 +653,26 @@ import Foundation
     }
     return nil
   }
+
+  private func startAndObserveUploadTask(task: StorageUploadTask,
+                                         completion: ((_: StorageMetadata?, _: Error?) -> Void)?) {
+    if let completion = completion {
```

Comment:
I realize this is not a behavior change from the previous code, but should we enqueue the task regardless of whether or not there's an observation callback? As the code currently is, if a user tries an upload with no observers, the upload will never actually begin.

Reply:
It does get called whether or not there is a completion handler.

The subsequent braces are end of function and end of class. 

The end of class one makes it confusing...